pkg/util/steps/refreshing.go

```diff
@@ -84,7 +86,23 @@ func (s *authorizationRefreshingActionStep) run(ctx context.Context, log *logrus
 		return true, err
 	}, timeoutCtx.Done())
 
-	return err
+	// After timeout, return any actionable errors to the user
+	switch err != nil {
+	case azureerrors.IsUnauthorizedClientError(err):
+		return s.servicePrincipalCloudError(
+			"The provided service principal application ID was not found in the tenant. Please ensure that the provided clientID and client secret are correct.",
+		)
+	case azureerrors.HasAuthorizationFailedError(err):
+		return s.servicePrincipalCloudError(
+			"Authorization using provided credentials failed. Please ensure that the provided clientID and client secret are correct.",
+		)
+	case azureerrors.IsInvalidSecretError(err):
+		return s.servicePrincipalCloudError(
+			"Invalid client secret provided. Please ensure that the provided clientID and client secret are correct.",
+		)
+	}
```

Comment:
I think one of these is legacy from before we were using MSGraph.  I think they're related to the same error message though.  I would just say: `Invalid Client secret`.  May want to rope in Jerome, Oren, or Puneet for customer-facing message though.  I'll ping them.

Reply:
So just to confirm, we want to change the error message from `"Invalid client secret provided. Please ensure that the provided clientID and client secret are correct."` to just `"Invalid client secret"?

Reply:
I changed this to combine these two, and use the error messages recommended by Puneet 👍